python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -447,6 +449,11 @@ def __new__(
             metadata=normalize_metadata(check.opt_mapping_param(metadata, "metadata")),
         )
 
+    @public
+    def get_asset_checks_def(self, key: AssetCheckKey) -> AssetChecksDefinition:
+        """Get a canonicalized :py:class:`AssetChecksDefinition` by key."""
+        return check.inst(self.get_asset_graph().assets_def_for_key(key), AssetChecksDefinition)
```

Comment:
this one is a little weird for me -- asset checks can be defined within AssetsDefinitions, so it's pretty hard to know in advance if this check.inst will error.

would prefer to leave this one off, especially as right now it doesn't have the `preview` decorator